### ISSUE TITLE
p-tree: selection propagation for selectionMode="checkbox" does not work with virtual scrolling

### ISSUE THREAD

**Issue by randes73** [primefaces/primeng#12608]
### Describe the bug

Please note that this is a duplicate of the following but that was closed and has not been re-opened:
https://github.com/primefaces/primeng/issues/11760

Selection propagation does not work as expected when p-tree configured with selectionMode="checkbox" and virtualScroll="true". The behavior can be observed using the stackblitz example link, adding virtualScroll="true" to the primeNG p-tree virtual scroll example, or using something similar to the following with some hierarchical data:
```
<p-tree
  [value]="files3"
  selectionMode="checkbox"
  [virtualScroll]="true"
  [virtualNodeHeight]="33"
  scrollHeight="220px"
></p-tree>
```

### Environment

Angular 14.2 using chromium based browser (Edge or Chrome)

### Reproducer

https://stackblitz.com/edit/github-zqg5s9?file=src/app/app.component.html

### Angular version

14.2.12

### PrimeNG version

14.1.2

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

16.17.0

### Steps to reproduce the behavior

Select and/or deselect child nodes using p-tree or p-treeselect with virtual scrolling. Parent node selection or partial selection is unaffected.

### Expected behavior

- When all child nodes are selected and the user deselects one of the child nodes, the parent should be partially selected
- When one or more child nodes are selected and the user deselects all of the child nodes, the parent should also be deselected
- When the parent is selected by the user, all child nodes should be selected
- When the parent is deselected by the user, all child nodes should be deselected
- When some child nodes are selected but not all child nodes and the user selects all of the child nodes, the parent should also be selected

**Comment by randes73** [comment]
Hello @samil90 ... no, I haven't received any updates.  The prior ticket, #11760, was closed by admin stating that I should try the latest version at the time.  I had asked for the previous ticket to be re-opened but after receiving no response on that request, I opened this new ticket after verifying the issue still existed.  

I have just updated the stackblitz below verifying that the behavior still exists as of 17.0.0-beta.1:
https://stackblitz.com/edit/github-zqg5s9-byjglq?file=src%2Fapp%2Fapp.component.html

We have had to avoid using virtual scrolling where the desired selection propagation is desired within our app.  We have been lucky so far that performance hasn't been an issue.
